code/modules/reagents/chemistry/holder.dm

```diff
@@ -577,12 +577,27 @@
 				var/check = reaction_check(A, R)
 				if(!check)
 					continue
+				if(handle_exotic_blood(A, R, method, volume_modifier))
+					continue
 				R.reaction_mob(A, method, R.volume * volume_modifier, show_message)
 			if("TURF")
 				R.reaction_turf(A, R.volume * volume_modifier, R.color)
 			if("OBJ")
 				R.reaction_obj(A, R.volume * volume_modifier)
 
+/datum/reagents/proc/handle_exotic_blood(atom/A, datum/reagent/R, method, volume_modifier)
+	if(!R || !method || method != REAGENT_INGEST)
+		return FALSE
+	if(ishuman(A))
+		var/mob/living/carbon/human/H = A
+		if(R.id == H.dna.species.exotic_blood)
+			//this is the same formula from the species handle_reagent proc
```

Comment:
redundant comment refers to deleted code.

Reply:
removed